### PR TITLE
Add tests for edit internship and project

### LINKED ISSUE #87: Re-add tests for EditCommand
I suppressed the EditCommand test in #86 as it involved creating some `EditXXXDescriptorBuilder` classes. 

It may take a bit of time to do and I wasn't sure whether we are going with the current `Descriptor`s.

### PR DESCRIPTION
**Pull request by nhamhung**
One part I'm not sure how to test: If we edit an item into another with the same name in the item list -> then DuplicateItemException should be thrown.

But I did this test in integration so it should be fine.

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/edit/EditInternshipCommandTest.java [hunk 1]
```diff
@@ -1,2 +1,182 @@
...
+    }
+
+    @Test
+    public void execute_duplicateInternship_throwsCommandException() throws CommandException {
+        Index validIndex = INDEX_FIRST_ITEM;
+        setEditInternshipDescriptor();
+        EditInternshipCommand editCommand = new EditInternshipCommand(validIndex, editInternshipDescriptor);
+        ModelStubContainingInternshipEdited modelStub = new ModelStubContainingInternshipEdited(
+                new InternshipBuilder().build());
+        modelStub.addInternship(sampleEditedInternship);
```
Comment (nhamhung) [inline src/test/java/seedu/address/logic/commands/edit/EditInternshipCommandTest.java]: This part i tried assertThrows exception but it does not throw and the modelStub just edit the first Internship into a duplicate of the second Internship

Comment (nhamhung) [reply src/test/java/seedu/address/logic/commands/edit/EditInternshipCommandTest.java]: And I don't include this @Test in Skill, Project and Resume yet! Just put here as a consideration

File: src/main/java/seedu/address/logic/commands/edit/EditSkillDescriptor.java [hunk 2]
```diff
@@ -78,6 +78,7 @@ public boolean equals(Object other) {
         EditSkillDescriptor e = (EditSkillDescriptor) other;
 
         return getName().equals(e.getName())
+                && getLevel().get().getLevelCode() == e.getLevel().get().getLevelCode()
```
Comment (helloImHai) [inline src/main/java/seedu/address/logic/commands/edit/EditSkillDescriptor.java]: This line is not safe bc get() can return null

File: src/test/java/seedu/address/logic/commands/edit/EditNoteCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+import seedu.address.testutil.EditNoteDescriptorBuilder;
+import seedu.address.testutil.ModelStub;
+import seedu.address.testutil.NoteBuilder;
+
+public class EditNoteCommandTest {
+    private Note sampleEditedNote = FINISH_HOMEWORK;
+
+    @Test
+    public void constructor_nullNote_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new EditInternshipCommand(null, null));
```
Comment (helloImHai) [inline src/test/java/seedu/address/logic/commands/edit/EditNoteCommandTest.java]: EditNoteCommand

File: src/test/java/seedu/address/testutil/EditProjectDescriptorBuilder.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    public EditProjectDescriptorBuilder() {
+        descriptor = new EditProjectDescriptor();
+    }
+
+    public EditProjectDescriptorBuilder(EditProjectDescriptor descriptor) {
+        this.descriptor = new EditProjectDescriptor(descriptor);
+    }
+
+    /**
+     * Returns an {@code EditInternshipDescriptor} with fields containing {@code internship}'s details
```
Comment (helloImHai) [inline src/test/java/seedu/address/testutil/EditProjectDescriptorBuilder.java]: Project

File: src/test/java/seedu/address/testutil/EditResumeDescriptorBuilder.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    public EditResumeDescriptorBuilder() {
+        descriptor = new EditResumeDescriptor();
+    }
+
+    public EditResumeDescriptorBuilder(EditResumeDescriptor descriptor) {
+        this.descriptor = new EditResumeDescriptor(descriptor);
+    }
+
+    /**
+     * Returns an {@code EditInternshipDescriptor} with fields containing {@code internship}'s details
```
Comment (helloImHai) [inline src/test/java/seedu/address/testutil/EditResumeDescriptorBuilder.java]: Resume

File: src/test/java/seedu/address/logic/commands/edit/EditProjectCommandTest.java [hunk 6]
```diff
@@ -1,4 +1,148 @@
...
+
+        assertEquals(String.format(EditProjectCommand.MESSAGE_EDIT_PROJECT_SUCCESS,
+                sampleEditedProject.getName().fullName),
+                commandResult.getFeedbackToUser());
+    }
+
+    @Test
+    public void create_withNullDescriptorField_editedProject() {
+        EditProjectDescriptor editProjectDescriptor = new EditProjectDescriptor();
+        editProjectDescriptor.setTags(null);
```
Comment (helloImHai) [inline src/test/java/seedu/address/logic/commands/edit/EditProjectCommandTest.java]: Might want to remove this?

**Review by helloImHai — CHANGES_REQUESTED** [review]
Nice one bro 🥇
Give the document a quick glance for typos, remove the equals(), and remove the setABC(null) and I think it's ready to merge

**Comment by nhamhung** [comment]
> One part I'm not sure how to test: If we edit an item into another with the same name in the item list -> then DuplicateItemException should be thrown.

**Review by chrisjwelly — COMMENTED** [review]
LGTM! :tada:

But gotta make Hai's fixes first

**Review by helloImHai — APPROVED** [review]
All fine and dandy 💃💃

**Pull request merged**